src/library_webgpu.js

```diff
@@ -912,7 +912,7 @@ var LibraryWebGPU = {
     var viewFormatCount = {{{ gpu.makeGetU32('descriptor', C_STRUCTS.WGPUTextureDescriptor.viewFormatCount) }}};
     if (viewFormatCount) {
       var viewFormatsPtr = {{{ makeGetValue('descriptor', C_STRUCTS.WGPUTextureDescriptor.viewFormats, '*') }}};
-      desc["viewFormats"] = Array.from({{{ makeHEAPView(`${POINTER_BITS}`, 'viewFormatsPtr', `viewFormatsPtr + viewFormatCount * ${POINTER_SIZE}`) }}},
+      desc["viewFormats"] = Array.from({{{ makeHEAPView(`32`, 'viewFormatsPtr', `viewFormatsPtr + viewFormatCount * 4`) }}},
```

Comment:
re: testing, I think we could patch a test into this existing test:
https://github.com/emscripten-core/emscripten/blob/4765f9c36f3706b37d295627c44307689a88bad7/test/webgpu_basic_rendering.cpp#L308

```diff
         wgpu::TextureDescriptor descriptor{};
         descriptor.usage = wgpu::TextureUsage::RenderAttachment | wgpu::TextureUsage::CopySrc;
         descriptor.size = {1, 1, 1};
         descriptor.format = wgpu::TextureFormat::BGRA8Unorm;
+        // Test for viewFormats binding
+        std::array<wgpu::TextureFormat, 2> viewFormats =
+            { wgpu::TextureFormat::BGRA8Unorm, wgpu::TextureFormat::BGRA8Unorm };
+        descriptor.viewFormatCount = viewFormats.size();
+        descriptor.viewFormats = viewFormats.data();
         readbackTexture = device.CreateTexture(&descriptor);
```

or something like that. The resulting JS call will be:

> ```js
> // before bugfix, throws TypeError:
> d.createTexture({ size: [4, 4], usage: GPUTextureUsage.COPY_SRC, format: 'bgra8unorm',
>     viewFormats: [undefined, undefined /* or possibly something else due to buffer overrun */] });
> // after bugfix, OK:
> d.createTexture({ size: [4, 4], usage: GPUTextureUsage.COPY_SRC, format: 'bgra8unorm',
>     viewFormats: ['bgra8unorm', 'bgra8unorm'] });
> ```

Unfortunately it wouldn't be quite so easy to test the requiredFeatures one.